src/sparse_int_set.jl

```diff
@@ -0,0 +1,292 @@
+    i <= 0 && throw(DomainError("Only positive Ints allowed."))
+    pageid, offset = pageid_offset(s, i)
+    page, newly_created = assure!(s, pageid)
+    if newly_created || page[offset] == 0
+        @inbounds page[offset] = length(s) + 1
+        @inbounds s.counters[pageid] += 1
+        push!(s.packed, i)
+        return s
+    end
+    return s
+end
+push!(s::SparseIntSet, is::Integer...) = (for i in is; push!(s, i); end; return s)
+
+function dirty_pop!(s::SparseIntSet)
+    if isempty(s)
+        throw(ArgumentError("Cannot pop an empty set."))
+    end
+    id = pop!(s.packed)
+    pageid, offset = pageid_offset(s, id)
+    @inbounds s.reverse[pageid][offset] = 0
+    @inbounds s.counters[pageid] -= 1
+    return id
+end
+
+@propagate_inbounds function pop!(s::SparseIntSet)
+    id = dirty_pop!(s::SparseIntSet)
+    cleanup!(s)
+    return id
+end
+
+function dirty_pop!(s::SparseIntSet, id::Integer)
+    id < 0 && throw(ArgumentError("Int to pop needs to be positive."))
+
+    @boundscheck if !in(id, s)
+        throw(BoundsError(s, id))
+    end
+    @inbounds begin
+        packed_endid           = s.packed[end] 
+        from_page, from_offset = pageid_offset(s, id)
+        to_page, to_offset     = pageid_offset(s, packed_endid)
+
+        packed_id                         = s.reverse[from_page][from_offset]
+        s.packed[packed_id]               = packed_endid
+        s.reverse[to_page][to_offset]     = s.reverse[from_page][from_offset]
+        s.reverse[from_page][from_offset] = 0
+        s.counters[from_page] -= 1
+        pop!(s.packed)
+    end
+    return id
+end
+
+@propagate_inbounds function pop!(s::SparseIntSet, id::Integer)
+    id = dirty_pop!(s, id)
+    cleanup!(s)
+    return id
+end
+
+function pop!(s::SparseIntSet, id::Integer, default)
+    id < 0 && throw(ArgumentError("Int to pop needs to be positive."))
+    in(id, s) ? (@inbounds pop!(s, id)) : default
+end
+popfirst!(s::SparseIntSet) = pop!(s, first(s))
+
+iterate(set::SparseIntSet, args...) = iterate(set.packed, args...) 
+
+last(s::SparseIntSet) = isempty(s) ? throw(ArgumentError("Empty set has no last element.")) : last(s.packed)
+
+union(s::SparseIntSet, ns) = union!(copy(s), ns)
+union!(s::SparseIntSet, ns) = (for n in ns; push!(s, n); end; s)
+
+intersect(s1::SparseIntSet) = copy(s1)
+intersect(s1::SparseIntSet, ss...) = intersect(s1, intersect(ss...))
+function intersect(s1::SparseIntSet, ns)
+    s = SparseIntSet()
+    for n in ns
+        n in s1 && push!(s, n)
+    end
+    return s
+end
+
+intersect!(s1::SparseIntSet, ss...) = intersect!(s1, intersect(ss...))
+
+#Is there a more performant way to do this?
+intersect!(s1::SparseIntSet, ns) = copy!(s1, intersect(s1, ns))
+
+setdiff(s::SparseIntSet, ns) = setdiff!(copy(s), ns)
+setdiff!(s::SparseIntSet, ns) = (for n in ns; pop!(s, n, nothing); end; s)
+
+function ==(s1::SparseIntSet, s2::SparseIntSet)
+    length(s1) != length(s2) && return false
+    return all(x -> in(x, s1), s2)
```

Comment:
Can use the single argument form of `in` for this:

```suggestion
    return all(in(s1), s2)
```